conda/plugins/types.py

```diff
@@ -347,3 +348,45 @@ class CondaRequestHeader:
 
     name: str
     value: str
+
+
+class EnvironmentSpecifierBase(ABC):
```

Comment:
I think we should use `EnvironmentSpec` since it's already called `RequirementsSpec` and `YamlFileSpec`.

Sure we can rename all of these but I'm pretty sure we are not about to rename `MatchSpec` so I think we should be consistent and stick to the `*Spec` convention.